format/moc3/moc3.go

```diff
@@ -0,0 +1,1766 @@
+package moc3
+
+// https://github.com/OpenL2D/moc3ingbird/blob/master/src/moc3.hexpat
```

Comment:
👍 interesting to see two implementations, are the comparable completnesswise? and as someone who has worked with both, how do they compare? the go code feels a bit more verbose?

Reply:
The Go code is largely autogenerated from the original implementation and then edited. They're at feature parity.

Some of the verbosity of the go decoder is probably due to some code duplication I can fix.